log4j-slf4j-impl/src/main/java/org/apache/logging/slf4j/Log4jMarkerFactory.java

```diff
@@ -36,9 +38,11 @@
     private static final Logger LOGGER = StatusLogger.getLogger();
 
     private final ConcurrentMap<String, Marker> markerMap = new ConcurrentHashMap<>();
+    private final Set<String> detachedMarkers = new CopyOnWriteArraySet<>();
```

Comment:
Doesn't tracking the markers defeat the purpose of them being "detached"?

Reply:
What about a `Map<String, WeakReference<Marker>>`?

Reply:
@ppkarwasz thank you for the review :) maybe the name of the field is misleading: should be "detachedMarkerNames"! For my use case, the marker name is always "STRUCTURED_ARG" but every instance has different data (in my use case, health check data). 

Basically, log4j is designed to use markers as "glorified strings", instead I would like to use them as "data carrier" (after my discussion with @rgoers, see below). I hope it is clear enough!